src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -1426,9 +1426,9 @@ public async Task WaitForExitAsync(CancellationToken cancellationToken = default
                 throw;
             }
 
-            var tcs = new TaskCompletionSourceWithCancellation<object>();
+            var tcs = new TaskCompletionSourceWithCancellation<bool>();
```

Comment:
Out of curiosity why was this needed?

Reply:
It needed to be made `object?` so that the `null` passed to `TrySetResult` was valid.  And as long as I was changing it, I changed it to just be a bool, which is what we generally use elsewhere in such situations where the T is actually just a dummy value.

Reply:
I see. Makes sense. Thanks for explaining.